### DISCUSSION TITLE
Running into difficulties rerouting system-wide traffic to a SOCKS5 proxy [Docker]

### DISCUSSION THREAD

**Discussion by optdxf** [xjasonlyu/tun2socks#204] [Q&A]
Hey - merry Christmas!

I'm trying to set up a systemwide proxy on Linux to reroute all traffic to a SOCKS5 proxy (with authentication) as an alternative to using `proxychains-ng`. I've tried two approaches to no avail, and was wondering if you could provide some guidance:

I tried following your Docker Compose example ([https://github.com/xjasonlyu/tun2socks/wiki/Use-docker-compose](https://github.com/xjasonlyu/tun2socks/wiki/Use-docker-compose)). Using the exact same docker compose file with default settings and setting `PROXY=socks5://<username>:<password>@<proxy_ip>:12324` (where I've omitted the username, password, and proxy IP), the output says `[STACK] tun://tun0 <-> socks5://<proxy_ip>:12324`. However, traffic doesn't seem to go through properly. For example, if I try and run `apk install curl`, I get the error message: 

```
[TCP] dial 199.232.146.133:443: connect to <proxy_ip>:12324: dial tcp <proxy_ip>:12324: connect: no route to host
```

A second approach I tried was just spinning up an Ubuntu container (via `sudo docker run -ti --cap-add=NET_ADMIN --device /dev/net/tun:/dev/net/tun ubuntu:latest`) and following the commands in this example ([https://github.com/xjasonlyu/tun2socks/wiki/Examples](https://github.com/xjasonlyu/tun2socks/wiki/Examples)). I tried running the following commands in the Docker container (which essentially follow the Linux example in the wiki):
```
apt-get upgrade -y && apt-get update -y && apt-get install -y curl wget iproute2 unzip && cd /home && wget https://github.com/xjasonlyu/tun2socks/releases/download/v2.4.1/tun2socks-linux-amd64-v3.zip && unzip tun2socks-linux-amd64-v3.zip
ip tuntap add mode tun dev tun0
ip addr add 198.18.0.1/15 dev tun0
ip link set dev tun0 up
ip route del default
ip route add default via 198.18.0.1 dev tun0 metric 1
ip route add default via 172.17.0.1 dev eth0 metric 10
sysctl net.ipv4.conf.all.rp_filter=0
sysctl net.ipv4.conf.eth0.rp_filter=0
nohup ./tun2socks-linux-amd64-v3 -device tun0 -proxy socks5://<username>:<password>@<proxy_ip>:12324 -interface eth0 &
```

Upon running something like `apt-get update`, which requires connecting to a server, I get the following tun2socks error:
```
[UDP] dial 185.12.64.2:53: client handshake: UDP ASSOCIATE: command not supported.
```

Do you have any guidance on how to go about just redirecting all traffic through a SOCKS5 proxy (that I can confirm by curl'ing some site that prints the public IPv4 address, which should match the proxy IP, for example)? I'm pretty new to networking, so I apologize if this is a simple thing.

**Comment by rsh10981** [comment]
I am trying to do the same thing. Were you able to resolve this? If so, can you share how?
